### ISSUE TITLE
Mirai error message outside of mirai function (Can't convert `NA` <unresolvedValue> to <integer>.)

### ISSUE THREAD

**Issue by vituri** [shikokuchuo/mirai#8]
Hi!

I've got this strange error where the mirai object seems to "pollute" the R environment and give errors in other places. For example, when running this piece of code:

```
library(dplyr)
library(mirai)

{
  x = mirai({
    iris %>%
      select(Species)
  }, iris = iris, count = count)
  x$data

  iris %>%
    select(Species)

}

iris %>%
  select(Species)

```
I got the error 
```
Error in `select()`:
! Can't convert `NA` <unresolvedValue> to <integer>.
Run `rlang::last_error()` to see where the error occurred.
```
and then, whenever I run just this piece:
```
iris %>%
  select(Species)
```
I got the same error. It seems that if I run an expression in mirai and this results in error, I cannot run this expression outside of mirai either.

My session info:
```
> sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=pt_BR.UTF-8       LC_NUMERIC=C               LC_TIME=pt_BR.UTF-8        LC_COLLATE=pt_BR.UTF-8     LC_MONETARY=pt_BR.UTF-8    LC_MESSAGES=pt_BR.UTF-8   
 [7] LC_PAPER=pt_BR.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=pt_BR.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mirai_0.4.1 dplyr_1.0.9

loaded via a namespace (and not attached):
 [1] fansi_1.0.3       assertthat_0.2.1  utf8_1.2.2        crayon_1.5.1      R6_2.5.1          DBI_1.1.2         lifecycle_1.0.1   magrittr_2.0.3    pillar_1.7.0     
[10] rlang_1.0.2       cli_3.3.0         rstudioapi_0.13   vctrs_0.4.1       generics_0.1.2    ellipsis_0.3.2    tools_4.2.0       glue_1.6.2        purrr_0.3.4      
[19] nanonext_0.5.0    compiler_4.2.0    pkgconfig_2.0.3   sessioninfo_1.2.2 tidyselect_1.1.2  tibble_3.1.7     
```

**Comment by shikokuchuo** [comment]
I will need some time to investigate... but if something like this is happening it is more likely than not a bug in `dplyr` or one of its dependencies as `select()` is known to use non-standard evaluation that may be catching something out of its proper scope.

Just in case you haven't figured it out, for the mirai to return correctly:

```
x <- mirai({
    library(dplyr)
    iris %>%
      select(Species)
  }, iris = iris)
```
or more succinctly:
```
x <- mirai(dplyr::select(x, Species), x = iris)
```

**Comment by shikokuchuo** [comment]
@vituri thanks for reporting. This was due to an issue in the nanonext package, which is now fixed (shikokuchuo/nanonext#3).

You can install the dev version of nanonext using
```
install.packages("nanonext", repos = "shikokuchuo.r-universe.dev")
```
or a new version should be released to CRAN in the next few days.
